From aeb1071733cda206d4911763636f9425153c3e2a Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <yasufum.o@gmail.com>
Date: Thu, 12 Sep 2019 18:18:00 +0900
Subject: [PATCH 9/9] shared: add TODO to fix bug in latency stats

Spp_vf and spp_mirror has a feature for measuring latency of ring PMD.
This optional feature is intended to be used by developers and is not
activate by default. It is activated by editing Makefile in src/vf and
src/mirror.

Although this feature is not interested by most of users and does not
prevent users from using SPP, it should be fixed. This update is to add
TODO comments for notifying users to not use this feature until the bug
is fixed.

Signed-off-by: Yasufumi Ogawa <yasufum.o@gmail.com>
---
 src/mirror/Makefile                                | 4 ++++
 src/shared/secondary/spp_worker_th/latency_stats.c | 5 +++++
 src/vf/Makefile                                    | 4 ++++
 3 files changed, 13 insertions(+)

diff --git a/src/mirror/Makefile b/src/mirror/Makefile
index 0adab15..0bd079a 100644
--- a/src/mirror/Makefile
+++ b/src/mirror/Makefile
@@ -42,6 +42,10 @@ CFLAGS += -DSPP_MIRROR_SHALLOWCOPY
 
 # Optional Settings
 #CFLAGS += -DSPP_DEMONIZE
+
+# TODO(Hideyuki Yamashita) ringlatencystats cannot run because a fatal
+# bug in initialization of this feature. So you should not use it until
+# this fatal bug is fixed.
 #CFLAGS += -DSPP_RINGLATENCYSTATS_ENABLE
 
 ifeq ($(CONFIG_RTE_BUILD_SHARED_LIB),y)
diff --git a/src/shared/secondary/spp_worker_th/latency_stats.c b/src/shared/secondary/spp_worker_th/latency_stats.c
index 33b3b44..d43f6a8 100644
--- a/src/shared/secondary/spp_worker_th/latency_stats.c
+++ b/src/shared/secondary/spp_worker_th/latency_stats.c
@@ -48,6 +48,11 @@ cycles_per_ns(void)
 	return rte_get_timer_hz() / NS_PER_SEC;
 }
 
+/**
+ * TODO(Hideyuki Yamashita) This function has a fatal bug in rte_zmalloc()
+ * for `g_stats_info` and should be fixed. rte_zmalloc() returns NULL for
+ * unknow reason.
+ */
 int
 sppwk_init_ring_latency_stats(uint64_t samp_intvl, uint16_t stats_count)
 {
diff --git a/src/vf/Makefile b/src/vf/Makefile
index 742986c..43e5e4d 100644
--- a/src/vf/Makefile
+++ b/src/vf/Makefile
@@ -37,6 +37,10 @@ CFLAGS += -DSPP_VF_MODULE
 
 # Optional Settings
 #CFLAGS += -DSPP_DEMONIZE
+
+# TODO(Hideyuki Yamashita) ringlatencystats cannot run because a fatal
+# bug in initialization of this feature. So you should not use it until
+# this fatal bug is fixed.
 #CFLAGS += -DSPP_RINGLATENCYSTATS_ENABLE
 
 ifeq ($(CONFIG_RTE_BUILD_SHARED_LIB),y)
-- 
2.17.1

